src/mlang/backend_compilers/bir_to_ocaml.ml

```diff
@@ -0,0 +1,387 @@
+        (Pos.unmark v.mir_var.name)
+        generate_local_defs local_defs
+        (Pos.unmark variable.mir_var.name)
+        (format_tgv_set_with_offset (get_var_pos variable) v)
+        tgv_expression
+  | InputVar -> assert false
+
+let generate_verif (oc : Format.formatter) (condition_data : Bir.condition_data)
+    =
+  let open Strings in
+  Format.fprintf oc "let verif_cond = %s in@,"
+    (let se, _ = generate_ocaml_expr condition_data.cond_expr in
+     se);
+  let cond_error, alias = condition_data.cond_error in
+  let error_name = sanitize_str cond_error.Mir.Error.name in
+  let error_kind, exception_raising =
+    match cond_error.Mir.Error.typ with
+    | Anomaly -> ("Anomaly", ";raise (M_exn context.errors)")
+    | Discordance -> ("Discordance", "")
+    | Information -> ("Information", "")
+  in
+  let error_kind_code = sanitize_str cond_error.Mir.Error.descr.kind in
+  let error_major_code = sanitize_str cond_error.Mir.Error.descr.major_code in
+  let error_minor_code = sanitize_str cond_error.Mir.Error.descr.minor_code in
+  let error_description = sanitize_str cond_error.Mir.Error.descr.description in
+  let error_alias =
+    match alias with
+    | Some v -> (
+        match (Bir.var_to_mir v).Mir.Variable.alias with
+        | Some alias -> "(( " ^ alias ^ " ))"
+        | None -> "")
+    | None -> ""
+  in
+  Format.fprintf oc
+    "match verif_cond with@,\
+     | { undefined = true ; value = _ }@,\
+     | { undefined = false ; value = 0.0 } -> ()@,\
+     | _ -> (@[<v 0> context.errors <- {@,\
+     name = \"%s\";@,\
+     kind = \"%s (%s)\";@,\
+     major_code = \"%s\";@,\
+     minor_code = \"%s\";@,\
+     description = \"%s\";@,\
+     alias = \"%s\"} :: context.errors@,\
+     @])%s@,"
+    error_name error_kind error_kind_code error_major_code error_minor_code
+    error_description error_alias exception_raising
+
+let generate_rov_header (oc : Format.formatter) (rov : Bir.rule_or_verif) : unit
+    =
+  let tname = match rov.rov_code with Rule _ -> "rule" | Verif _ -> "verif" in
+  Format.fprintf oc "m_%s_%s context" tname (Pos.unmark rov.rov_name)
+
+let rec generate_stmts (program : Bir.program) (oc : Format.formatter)
+    (stmts : Bir.stmt list) : unit =
+  Format.pp_print_list ~pp_sep:pp_statement_separator (generate_stmt program) oc
+    stmts
+
+and generate_stmt (program : Bir.program) (oc : Format.formatter)
+    (stmt : Bir.stmt) : unit =
+  match Pos.unmark stmt with
+  | SAssign (variable, variable_data) ->
+      generate_var_def variable variable_data oc
+  | SConditional (cond_expression, tt, ff) ->
+      let pos = Pos.get_position stmt in
+      let fname =
+        String.map
+          (fun c -> if c = '.' then '_' else c)
+          (Filename.basename (Pos.get_file pos))
+      in
+      let cond_name =
+        Format.asprintf "cond_%s_%d_%d_%d_%d" fname (Pos.get_start_line pos)
+          (Pos.get_start_column pos) (Pos.get_end_line pos)
+          (Pos.get_end_column pos)
```

Comment:
Is this for debugging purposes ? Seems a bit overkill for a shadowable name.

Reply:
I would say: "probably close to how it was done in other backends" and the only use I picture for it in the foreseeable future is helping generated code readability.